File: Striver -AtoZ/Heap/findKthLargest.cpp
#include <bits/stdc++.h>

using namespace std ;

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        int n = nums.size() ;
        priority_queue<int,vector<int>,greater<int>> minHeap ;

        for(auto it : nums){
            minHeap.push(it) ;
            if(minHeap.size() > k){
                minHeap.pop() ;
            }
        }        

        return minHeap.top() ;
    }
};

File: Striver -AtoZ/Heap/findKthLargest.rs
use std::collections::BinaryHeap;
use std::cmp::Reverse;

impl Solution {
    pub fn find_kth_largest(nums: Vec<i32>, k: i32) -> i32 {
        let mut minHeap = BinaryHeap::new() ;
        let n = nums.len() ;     
        
        for i in 0..n{
            minHeap.push(Reverse(nums[i])) ;
            if minHeap.len() > k {
                minHeap.pop() ;
            }
        }

        minHeap.peek()
    }
}